sound/effect.h

```diff
@@ -352,11 +352,21 @@ class Effect {
 
 #ifdef NO_REPEAT_RANDOM
   int last_;
+  int last_sub_;
 #endif
 
   int random_subid() {
     if (!sub_files_) return 0;
+#ifdef NO_REPEAT_RANDOM
+    int sel = rand() % sub_files_;
+    for (int i = 0; i < 3 && sel == last_sub_; i++) {
+      sel = clamp(sel + 1 - (rand() & 2), 0, sub_files_ - 1);
```

Comment:
we don't need this complicated logic for sub-files sine there is no need to try to select select a "nearby" value.
Instead we should just do sel = rand() % sub_files_; again.

Reply:
also, we may need special cases if sub_files_ is two.
And, we may want to avoid doing this if last_ is different from whatever file is getting selected this time.

Reply:
updated and accounting for >2 and selected_ = last_ now